### PR TITLE
Hailt disable user frontend new

### PR DESCRIPTION
**Pull request by lthai2000**

### CODE REVIEW

File: backend/API/Controllers/UsersController.cs [hunk 1]
```diff
@@ -125,7 +126,51 @@ public async Task<ActionResult<Response<CreateUserResponse>>> CreateUser([FromBo
...
+            return Ok(response);
+        }
+        catch (Exception exception)
+        {
+            return HandleException(exception);
+        }
+    }
+
+    [Authorize(UserRole.Admin)]
+    [HttpPut("disability")]
```
Comment (truongtrinhv) [inline backend/API/Controllers/UsersController.cs]: should be disable only. Need to double check naming convention for api

File: backend/Application/Services/UserService.cs [hunk 2]
```diff
@@ -222,6 +224,57 @@ public async Task<Response<GetListUsersResponse>> GetListAsync(GetListUsersReque
         return new Response<GetListUsersResponse>(true, response);
     }
 
+    public async Task<Response> IsAbleToDisableUser(Guid id)
+    {
+        var isAnyValidAssignment = await IsAnyValidAssignmentBelongsToUser(id);
```
Comment (truongtrinhv) [inline backend/Application/Services/UserService.cs]: isAnyValidAssignment shoud be hasValidAssignment

File: backend/Application/Services/UserService.cs [hunk 3]
```diff
@@ -222,6 +224,57 @@ public async Task<Response<GetListUsersResponse>> GetListAsync(GetListUsersReque
...
+        var user = await userRepository.GetAsync(u => !u.IsDeleted &&
+                                                      u.Id == request.Id &&
+                                                      u.Location == request.Location);
+
+        if (user == null)
+        {
+            return new Response(false, ErrorMessages.NotFound);
+        }
+
+        var isAnyValidAssignment = await IsAnyValidAssignmentBelongsToUser(request.Id);
```
Comment (truongtrinhv) [inline backend/Application/Services/UserService.cs]: isAnyValidAssignment shoud be hasValidAssignment

Comment (truongtrinhv) [reply backend/Application/Services/UserService.cs]: IsAnyValidAssignmentBelongsToUser => HasValidAssignment

File: backend/Application/Services/UserService.cs [hunk 4]
```diff
@@ -222,6 +224,57 @@ public async Task<Response<GetListUsersResponse>> GetListAsync(GetListUsersReque
...
+
+    private async Task<bool> IsAnyValidAssignmentBelongsToUser(Guid userId)
+    {
+        var assignmentRepository = UnitOfWork.AsyncRepository<Assignment>();
+
+        var assignments = await assignmentRepository.ListAsync(a => !a.IsDeleted &&
+                                                                    a.AssignedTo == userId &&
+                                                                    a.State != AssignmentState.Declined);
+
+        return assignments.Any();
```
Comment (truongtrinhv) [inline backend/Application/Services/UserService.cs:275]: This code   var assignments = await assignmentRepository.ListAsync(a => !a.IsDeleted &&
                                                                    a.AssignedTo == userId &&
                                                                    a.State != AssignmentState.Declined);

        return assignments.Any();

        should be shorter like


    return await assignmentRepository.Any(a => !a.IsDeleted &&
                                                                    a.AssignedTo == userId &&
                                                                    a.State != AssignmentState.Declined);

File: backend/Application/Application.csproj [hunk 5]
```diff
@@ -18,4 +18,8 @@
     <ProjectReference Include="..\Infrastructure\Infrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="DTOs\Users\DisableUser\" />
```
Comment (truongtrinhv) [inline backend/Application/Application.csproj:22]: Folder name should be UserDisable

File: backend/Application.UnitTests/ServiceTests/UserServiceTests.cs [hunk 6]
```diff
@@ -443,7 +445,164 @@ public async Task ChangePasswordAsync_FirstTimeLoginAndValidInputs_ReturnsSucces
...
+    public async Task IsAbleToDisableUser_ValidAssignmentsExist_ReturnsNotSuccessResponse()
+    {
+        var assignmentRepository = new Mock<IAsyncRepository<Assignment>>();
+
+        _unitOfWork
+            .Setup(uow => uow.AsyncRepository<Assignment>())
+            .Returns(assignmentRepository.Object);
+
+        assignmentRepository
+            .Setup(ar => ar.ListAsync(
```
Comment (truongtrinhv) [inline backend/Application.UnitTests/ServiceTests/UserServiceTests.cs]: This code should as my comment in UserService

File: frontend/src/Pages/AdminPage/ManageUserPage/DetailedInfoUserPage.jsx [hunk 7]
```diff
@@ -62,7 +67,9 @@ export function DetailedInfoUserPage() {
               </tr>
               <tr>
                 <td className="font-bold">Location:</td>
-                <td>{data.location === "HaNoi" ? "Ha Noi" : "Ho Chi Minh City"}</td>
+                <td>
+                  {data.location === "HaNoi" ? "Ha Noi" : "Ho Chi Minh City"}
```
Comment (truongtrinhv) [inline frontend/src/Pages/AdminPage/ManageUserPage/DetailedInfoUserPage.jsx]: This string should to constant

File: frontend/src/Pages/AdminPage/ManageUserPage/DisableUserPage.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  const onCancel = () => {
+    navigate(-1);
+  };
+
+  return (
+    <>
+      {isAbleToDisable ? (
+        <Modal open={isModalOpen} closable={false} footer={false} width={400}>
```
Comment (truongtrinhv) [inline frontend/src/Pages/AdminPage/ManageUserPage/DisableUserPage.jsx:45]: Why did yous set to false here?

Comment (lthai2000) [reply frontend/src/Pages/AdminPage/ManageUserPage/DisableUserPage.jsx:45]: Because in design there is no close button

Comment (lthai2000) [reply frontend/src/Pages/AdminPage/ManageUserPage/DisableUserPage.jsx:45]: And I don't want to use the default footer of antd so I set the footer to false

**Review by truongtrinhv — APPROVED** [review]

**Pull request merged**